### ISSUE TITLE
k3s-serving expired x509: certificate has expired or is not yet valid

### ISSUE THREAD

**Issue by TrueCarry** [k3s-io/k3s#3144]
**Environmental Info:**
K3s Version: 
k3s version v1.18.8+k3s1 (6b595318)

Node(s) CPU architecture, OS, and Version:
Linux 99f9c510028d 4.4.0-116-generic #140-Ubuntu SMP Mon Feb 12 21:23:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
1 node local rancher cluster

**Describe the bug:**
While certificates in k3s/server/tls were updated, k3s-serving certificate in cluster expired.
![image](https://user-images.githubusercontent.com/5431520/113503907-74cbf780-953d-11eb-82db-585db06abed1.png)

**Steps To Reproduce:**

I don't know how to reproduce. I didn't install k3s myself. I've installed rancher in docker. First v2.2.1, then v2.4.2, now v2.5.7. Only flag I used in rancher is --no-cacerts.

**Expected behavior:**
Secret is updated before expiring.

**Actual behavior:**
Secret expired.

**Additional context / logs:**

```
Get "https://127.0.0.1:6443/version?timeout=15m0s": x509: certificate has expired or is not yet valid: current time 2021-04-04T09:04:37Z is after 2021-04-03T12:53:59Z
```

Is it safe to delete this secret? Will new one be generated on restart?

**Comment by TrueCarry** [comment]
After further reading comments and searching by "dynamic-cert.json" I've found brandond comment saying that removing it with secret should be okay. So I did and it's renewed now. It also seems that this problem is already known and fixed, so I'm closing issue.
```
rm /var/lib/rancher/k3s/server/tls/dynamic-cert.json
kubectl delete secret -n kube-system k3s-serving
```

**Comment by micw** [comment]
I ran into the same issue. On an older cluster (2.4.8), the k3s-serving cert expired. I changed the clock to make the cluster start again and upgraded to 2.5.7. Cluster starts fine but the cert still does not get renewed. Your workaround might work but I'd like to have the issue re-opened since it's not resolved.

**Comment by TrueCarry** [comment]
If someone else stumbles upon this issue, you most likely don't need to set time back to fix it, since you can use --insecure-skip-tls-verify=true on kubectl inside rancher container to delete secret. In my case setting time back broke half of workloads, so it was pretty painful.

**Comment by theAkito** [comment]
Happening on Rancher 2.5.7, too.

**Comment by neozeed** [comment]
So I guess the commands were ....?

```
docker exec -it docker_container -c "rm -rf /var/lib/rancher/k3s/server/tls/dynamic-cert.json"
docker exec -it docker_container -c "k3s kubectl --insecure-skip-tls-verify=true  delete secrets -n kube-system k3s-serving"
docker exec -it docker_container -c "k3s kubectl --insecure-skip-tls-verify=true  delete secrets -n cattle-system serving-cert"
docker restart docker_container
```

but of course......

```
# docker logs docker_container | head -10
2021-09-27 08:23:09.251272 W | etcdserver: read-only range request "key:\"/registry/services/endpoints/kube-system/kube-scheduler\" " with result "range_response_count:1 size:442" took too long (148.242744ms) to execute
2021-09-27 08:23:11.234593 I | http: TLS handshake error from 127.0.0.1:57008: remote error: tls: bad certificate
2021-09-27 08:23:13.236046 I | http: TLS handshake error from 127.0.0.1:57020: remote error: tls: bad certificate
2021/09/27 08:23:11 [INFO] Waiting for server to become available: Get https://127.0.0.1:6443/version?timeout=30s: x509: certificate has expired or is not yet valid
2021-09-27 08:23:15.237880 I | http: TLS handshake error from 127.0.0.1:57032: remote error: tls: bad certificate
2021-09-27 08:23:17.239101 I | http: TLS handshake error from 127.0.0.1:57048: remote error: tls: bad certificate
2021/09/27 08:23:13 [INFO] Waiting for server to become available: Get https://127.0.0.1:6443/version?timeout=30s: x509: certificate has expired or is not yet valid
2021-09-27 08:23:19.243370 I | http: TLS handshake error from 127.0.0.1:57068: remote error: tls: bad certificate
2021/09/27 08:23:15 [INFO] Waiting for server to become available: Get https://127.0.0.1:6443/version?timeout=30s: x509: certificate has expired or is not yet valid
2021/09/27 08:23:17 [INFO] Waiting for server to become available: Get https://127.0.0.1:6443/version?timeout=30s: x509: certificate has expired or is not yet valid
2021-09-27 08:23:21.246322 I | http: TLS handshake error from 127.0.0.1:57078: remote error: tls: bad certificate
```

that didn't do anything....

**Comment by brandond** [comment]
For anyone running into this in the rancher/rancher docker image, updating to a newer version of the docker image that contains a fixed version of K3s should resolve this automatically.
